src/system/SystemLayerImplSelect.cpp

```diff
@@ -608,6 +610,32 @@ SocketEvents LayerImplSelect::SocketEventsFromFDs(int socket, const fd_set & rea
     return res;
 }
 
+#if !CHIP_SYSTEM_CONFIG_USE_DISPATCH
+enum : intptr_t
```

Comment:
why is the state a intptr_t ? why not plain int or named enum?

Reply:
Once this decision is done, we have `auto` and `decltype` everywhere even though it is not obvious that a `GetState`-like call results in a intptr_t

Reply:
It's not a named enum because it's really up to the specific implementation class what gets stored there (though at the moment there's only one implementation class). In practice there aren't going to be a large number of these handlers, and due to alignment the memory cost of an int or an enum are going to be the same anyway. Using an intptr_t gives the implementation the ability to store a pointer if necessary.